src/storage/disk_manager.c

```diff
@@ -5917,19 +5880,28 @@ disk_dump_goodvol_all (THREAD_ENTRY * thread_p, INT16 volid, void *arg)
 STATIC_INLINE bool
 disk_is_valid_volid (VOLID volid)
 {
-  // a passive transaction server (which is, implicitly, a transaction server
-  // with remote storage) maintains a separate bookkeeping of permanent data volumes
-  // which is updated when executing corresponding recovery replication functions
-  // as part of the regular replication; this way, it is able to provide a valid
-  // answer via this function
-  const bool is_valid_perm_volid = is_tran_server_with_remote_storage ()?
-    (volid < disk_Page_server_perm_volume_count) : (volid < disk_Cache->nvols_perm);
+  if (is_tran_server_with_remote_storage () && is_passive_transaction_server ())
```

Comment:
second function implies the first; so only first condition is needed

Reply:
Right. However, I know that the first condition includes ATS and PTS.

ATS doesn't have to come into this part, because ATS updates disk_Cache well every time.
So, I think it's better to use only the second condition.

Reply:
@joohok As I know, the result of the is_passive_transaction_server() function is not guaranteed to use remote storage.

Reply:
@hornetmj  Yes, you are right. 
I was confused for a while whether PTS could operate as a single node or not.

Reply:
@joohok The passive_tran_server::uses_remote_storage() always returns true.

Reply:
Yes, I reduce the number of conditions. Thanks